map.go

```diff
@@ -1499,24 +1527,59 @@ type MapIterator struct {
 	err               error
 }
 
-func newMapIterator(target *Map) *MapIterator {
-	return &MapIterator{
+// keylessMapIterator iterates a Map defined with only values.
+// (Queue and Stack)
+//
+// See Map.Iterate.
+type keylessMapIterator struct {
+	target *Map
+	err    error
+}
+
+// newMapIterator return the correct MapIterator implementation
+// based on the underlying map type.
+func newMapIterator(target *Map) MapIterator {
+	if target.typ.isQueueStack() {
+		return &keylessMapIterator{
+			target: target,
+		}
+	}
+	return &keyValueMapIterator{
 		target:     target,
 		maxEntries: target.maxEntries,
 	}
 }
 
-// Next decodes the next key and value.
-//
-// Iterating a hash map from which keys are being deleted is not
-// safe. You may see the same key multiple times. Iteration may
-// also abort with an error, see IsIterationAborted.
+// next decodes the next value from a map of type Queue or Stack.
 //
 // Returns false if there are no more entries. You must check
 // the result of Err afterwards.
+func (mi *keylessMapIterator) next(_, valueOut interface{}) bool {
```

Comment:
I'd make this `MapIterator.pop()` and make `MapIterator.Next()` call `next()` or `pop()` depending on the outcome of `isQueueStack()`.